src/hotspot/share/utilities/spinCriticalSection.hpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_UTILITIES_SPINCRITICALSECTION_HPP
+#define SHARE_UTILITIES_SPINCRITICALSECTION_HPP
+
+#include "runtime/javaThread.hpp"
+
+class SpinCriticalSectionHelper {
+  friend class SpinCriticalSection;
+  template<class Lambda, class...Args>
+  friend class SpinSingleSection;
+  // Low-level leaf-lock primitives used to implement synchronization.
+  // Not for general synchronization use.
+  static void spin_acquire(volatile int* Lock);
+  static void spin_release(volatile int* Lock);
+  static bool try_spin_acquire(volatile int* Lock);
+};
+
+// Short critical section. To be used when having a
+// mutex is considered to be expensive.
```

Comment:
I think this is a really poor description, as I think it will encourage the use of these facilities in
inappropriate places.  Spin-lock usage ought to be pretty rare!  Note that the existing mechanism
is described as "Not for general synchronization use." I think better motivation is needed. Note
I'm not suggesting that doesn't exist, rather than motivation and usage guidelines should be
documented here. The comment for SpinCriticalSectionHelper in the .cpp file is the kind of
thing I'm looking for. I shouldn't have to look at that internal helper's implementation to find
such guidance.